src/App.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+	<router-view />
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import WalletStore from "./store/WalletStore";
+import { WalletKind } from "./types";
+declare var AlgoSigner: any; // eslint-disable-line
+
+export default defineComponent({
+	name: "App",
+	components: {},
+	data() {
+		return {
+			loading: true,
+		};
+	},
+	setup() {
+		const counter = WalletStore();
+
+		return {
+			setHasAlgoSigner: counter.setHasAlgoSigner,
+		};
+	},
+	mounted() {
+		this.checkAlgoSigner();
+		setTimeout(() => (this.loading = false), 200);
+	},
+	methods: {
+		async checkAlgoSigner() {
+			if (typeof AlgoSigner !== "undefined") {
+				this.setHasAlgoSigner(WalletKind.AlgoSigner);
+				await AlgoSigner.connect();
+			} else {
+				setTimeout(() => this.checkAlgoSigner(), 1);
```

Comment:
checking every 1ms way too ofthen, will burn the CPU. Let's change to 500ms. Also we should do maximum 5times. If it's not detected in 2.5s, then it means the AlgoSigner is not there - so we should send a notification to the user.